addon/components/flexberry-file.js

```diff
@@ -1,334 +1,523 @@
 
   /**
-   * Flag: indicates whether remove button is enabled now.
+    Flag: indicates whether remove button is enabled now.
+
+    @property removeButtonIsEnabled
+    @type Boolean
+    @readonly
    */
-  removeButtonIsEnabled: Ember.computed('uploadIsInProgress', 'downloadIsInProgress', 'value', function() {
-    var uploadIsInProgress = this.get('uploadIsInProgress');
-    var downloadIsInProgress = this.get('downloadIsInProgress');
-    var jsonValue = this.get('jsonValue');
+  removeButtonIsEnabled: Ember.computed('uploadIsInProgress', 'value', function() {
+    let uploadIsInProgress = this.get('uploadIsInProgress');
+    let jsonValue = this.get('jsonValue');
 
-    return !(uploadIsInProgress || downloadIsInProgress || Ember.isNone(jsonValue));
+    return !(uploadIsInProgress || Ember.isNone(jsonValue));
   }),
 
   /**
-   * Flag: indicates whether to upload file on 'relatedModel' 'preSave' event.
+    Flag: indicates whether to upload file on 'relatedModel' 'preSave' event.
+
+    @property uploadOnModelPreSave
+    @type Boolean
+    @default true
    */
   uploadOnModelPreSave: undefined,
 
   /**
-   * Flag: indicates whether to show upload button or not.
+    Flag: indicates whether to show preview element for images or not.
+
+    @property showPreview
+    @type Boolean
+    @default false
+   */
+  showPreview: false,
+
+  /**
+    Preview options change handler.
+   */
+  previewOptionsDidChange: Ember.on('init', Ember.observer('showPreview', 'selectedFile', 'jsonValue.previewUrl', function() {
+    if (!this.get('showPreview') || !Ember.isBlank(this.get('_previewImageAsBase64String'))) {
+      return;
+    }
+
+    let file = this.get('selectedFile');
+    if (!Ember.isNone(file)) {
+      let reader = new FileReader();
+      reader.onload = (e) => {
+        this.set('_previewImageAsBase64String', e.target.result);
+        this.set('previewDownloadIsInProgress', false);
+      };
+
+      this.set('previewDownloadIsInProgress', true);
+      reader.readAsDataURL(file);
+
+      return;
+    }
+
+    let previewUrl = this.get('jsonValue.previewUrl');
+    if (!Ember.isBlank(previewUrl)) {
+      // Download file preview.
+      this.set('previewDownloadIsInProgress', true);
+      Ember.$.ajax(previewUrl).done((data, textStatus, jqXHR) => {
+        this.set('_previewImageAsBase64String', data);
+      }).fail((jqXHR, textStatus, errorThrown) => {
+        this.showDownloadErrorModalDialog(this.get('jsonValue.fileName'), errorThrown);
+      }).always(() => {
+        this.set('previewDownloadIsInProgress', false);
+      });
+    }
+  })),
+
+  /**
+    Flag: indicates whether to show upload button or not.
+
+    @property showUploadButton
+    @type Boolean
+    @default false
    */
   showUploadButton: undefined,
 
   /**
-   * Flag: indicates whether to show download button or not.
+    Flag: indicates whether to show download button or not.
+
+    @property showDownloadButton
+    @type Boolean
+    @default true
```

Comment:
No. Some settings are still initializing through flexberry-base-component mechanism (from configuration/environment.js) and such setings need to be undefined in component's class.